components/sensor/ina2xx.rst

```diff
@@ -0,0 +1,199 @@
+
+.. code-block:: yaml
+
+    # Example configuration entry for I²C
+    i2c:
+      sda: D1
+      scl: D2
+        
+    sensor:
+      - platform: ina2xx_i2c
+        id: my_charge_sensor
+        model: INA228
+        address: 0x40
+        shunt_resistance: 0.010 ohm
+        max_current: 10 A
+        adc_range: 0
+        update_interval: 60s
+        current: "INA228 Current"
+        bus_voltage: "INA228 Bus Voltage"
+        charge: "INA228 Charge"
+
+
+Configuration variables:
+************************
+
+- **model** (**Required**, string): The model of the INA2xx sensor. Options are ``INA228``, ``INA237``, ``INA238``.
+- **address** (*Optional*, int): Manually specify the I²C address of the sensor. Defaults to ``0x40``.
+- **shunt_resistance** (**Required**, float): The value of the shunt resistor used for current calculation. No default value.
+- **max_current** (**Required**, float): The maximum current you are expecting. Component will use it to 
+  calibrate the sensor. No default value.
+- **adc_range** (*Optional*, ``0`` or ``1``): Selects the range for differential input across shunt
+  resistor. ``0`` for ±163.84 mV, ``1`` for ±40.96 mV range. Defaults to ``0``.
+- **adc_time** (*Optional*, :ref:`config-time`): The time in microseconds to perform a single ADC conversion. 
+  Defaults to ``4120 us``. Valid values are ``50 us``, ``84 us``, ``150 us``, ``280 us``, ``540 us``, 
+  ``1052 us``, ``2074 us``, ``4120 us``.
+
+  Instead of one time for all ADC measurements, separate configuration of conversion times for shunt voltage, 
+  bus voltage, and temperature measurements possible. Options are the same as for ``adc_time``.
+
+  - **bus_voltage** (*Optional*, :ref:`config-time`): Conversion time for bus voltage measurement.
+  - **shunt_voltage** (*Optional*, :ref:`config-time`): Conversion time for shunt voltage measurement.
+  - **temperature** (*Optional*, :ref:`config-time`): Conversion time for temperature measurement.
+
+- **adc_averaging** (*Optional*, integer): Selects ADC sample averaging count. Defaults to ``128``. 
+  Valid values are ``1``, ``4``, ``16``, ``64``, ``128``, ``256``, ``512``, ``1024``.
+- **temperature_coefficient** (*Optional*, integer from ``0`` to ``16383``): Temperature coefficient (ppm/°C) of the 
+  shunt for temperature compensation correction. Only applicable to INA228 and INA229 devices. Zero value means 
+  no compensation is done. Defaults to ``0``.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
+- All other options for I²C device as descibed in respective documentation.
+
+
+Over SPI
+--------
+
+Use ``ina2xx_spi`` component to connect INA229 or INA239 device over :ref:`SPI <spi>` bus. 
+
+
+.. code-block:: yaml
+
+    # Example configuration entry for SPI
+    spi:
+      clk_pin: D0
+      mosi_pin: D1
+      miso_pin: D2
+        
+    sensor:
+      - platform: ina2xx_spi
+        cs_pin: D3
```

Comment:
For pin numbers, we're now using this pattern (to force people to use their own pin(s)).
```suggestion
        cs_pin: GPIOXX
```